applauncher/004-three-by-three.qml

```diff
@@ -0,0 +1,320 @@
+                running: false
+                repeat: false
+                onTriggered: {
+                     if (!dragStopper) {
+                         parent.timedPressAndHold()
+                     }
+                     else {
+                         root.dragStop = false
+                     }
+                }
+            }
+
+            onClicked: {
+                clickToggle = true
+                model.object.launchApplication()
+            }
+
+            onPressed: {
+                pressAndHoldTimer.start();
+                root.dragStop = false
+            }
+
+            onPressedChanged: {
+                appTitle = model.object.title
+            }
+
+            onReleased: {
+                pressAndHoldTimer.stop();
+            }
+
+            onTimedPressAndHold: {
+                appTitle = model.object.title
+                clickToggle = false
+                pressToggle ? pressToggle = false : pressToggle = true
+                currentPressedIndex = index
+            }
+
+            Connections {
+                target: appLauncher
+                function onFakePressedChanged() {
+                    appTitle = model.object.title                   
+                }
+            }
+
+            Item {
+                id: circleWrapper
+
+                width: parent.width * .86
+                height: width
+                anchors {
+                    centerIn: parent
+                    verticalCenterOffset: -parent.height * .08
+                }
+
+                Rectangle {
+                    id: circle
+
+                    anchors.centerIn: parent
+                    width: parent.width
+                    height: width
+                    radius: width/2
+                    color: !pressAndHoldTimer.running ?
+                               launcherItem.pressed | fakePressed ?
+                                           alb.centerColor(launcherModel.get(currentPressedIndex).filePath) :
+                                           "#f4f4f4" : "#f4f4f4"
+                    Behavior on color {
+                        PropertyAnimation { target: circle; property: "color"; duration: 70 }
+                    }
+                }
+            }
+
+            DropShadow {
+                anchors.fill: circleWrapper
+                horizontalOffset: 0
+                verticalOffset: 0
+                radius: 8.0
+                samples: 17
+                color: "#80000000"
+                source: circleWrapper
+                cached: true
+            }
+
+            Icon {
+                id: icon
+
+                width: circleWrapper.width * .70
+                height: width
+                anchors.centerIn: circleWrapper
+                color: !pressAndHoldTimer.running ?
+                           launcherItem.pressed | fakePressed ?
+                                       "#ffffff" :
+                                       "#555555" : "#555555"
```

Comment:
These are overly complex ternary chains.
```
A ? B ? X : Y : Y
```
Is hard to read and should be equivalent to
```
A && B ? X : Y
```

Same comment for `circle.color`